src/azure-cli/azure/cli/command_modules/apim/_params.py

```diff
@@ -316,9 +316,9 @@ def load_arguments(self, _):
             'legal_terms',
             help="Product terms of use. Developers trying to subscribe to the product will be presented and required to accept these terms before they can complete the subscription process.")
         c.argument(
-            'subscription_required', options_list=['--subscription-required', '-s'],
+            'subscription_required', options_list=['--subscription-required', '-s'], arg_type=get_three_state_flag(),
             help="Whether a product subscription is required for accessing APIs included in this product.")
-        c.argument('approval_required', help="whether subscription approval is required. If false, new subscriptions will be approved automatically enabling developers to call the product’s APIs immediately after subscribing. If true, administrators must manually approve the subscription before the developer can any of the product’s APIs. Can be present only if subscriptionRequired property is present and has a value of false.")
+        c.argument('approval_required', arg_type=get_three_state_flag(), help="whether subscription approval is required. If false, new subscriptions will be approved automatically enabling developers to call the product’s APIs immediately after subscribing. If true, administrators must manually approve the subscription before the developer can any of the product’s APIs. Can be present only if subscriptionRequired property is present and has a value of false.")
```

Comment:
May I ask will the change of parameter type lead to breaking change to users?

Reply:
@arhaverly Could you answer this question? Please note that if you do not address this comment tomorrow, this PR will not catch up with the release of this sprint, and we will have to postpone it to the release of next sprint (**07-05**)

Reply:
This is to fix a bug that users currently encounter. This will not lead to a breaking change to the users.

Reply:
OK, I would like to confirm with you that the `--approval-required` parameter can only support the `True` or `False`, right? If other values are entered by users, an error will be raised?

Reply:
correct. an error will be raised if either `--approval-required` or `--subscription-required` do not have either `True` or `False` values. Ex:
az apim product create --resource-group andyhaverly --service-name andyhaverly-cli --product-name ex2 --subscription-required true --approval-required hello
az apim product create: 'hello' is not a valid value for '--approval-required'. Allowed values: true, false.